impeller/aiks/experimental_canvas.cc

```diff
@@ -44,6 +45,96 @@ static void ApplyFramebufferBlend(Entity& entity) {
   entity.SetBlendMode(BlendMode::kSource);
 }
 
+/// End the current render pass, saving the result as a texture, and then
+/// restart it with the backdrop cleared to the previous contents.
+///
+/// This method is used to set up the input for emulated advanced blends and
+/// backdrop filters.
+///
+/// Returns the previous render pass stored as a texture, or nullptr if there
+/// was a validation failure.
+static std::shared_ptr<Texture> FlipBackdrop(
+    std::vector<LazyRenderingConfig>& render_passes,
+    Point global_pass_position,
+    EntityPassClipStack& clip_coverage_stack,
+    ContentContext& renderer) {
+  auto rendering_config = std::move(render_passes.back());
+  render_passes.pop_back();
+
+  // If the very first thing we render in this EntityPass is a subpass that
+  // happens to have a backdrop filter or advanced blend, than that backdrop
+  // filter/blend will end may wind up sampling from the uninitialized texture
+  //
+  // By calling `pass_context.GetRenderPass` here, we force the texture to pass
+  // through at least one RenderPass with the correct clear configuration before
+  // any sampling occurs.
+  //
+  // In cases where there are no contents, we
+  // could instead check the clear color and initialize a 1x2 CPU texture
+  // instead of ending the pass.
+  rendering_config.inline_pass_context->GetRenderPass(0);
+  if (!rendering_config.inline_pass_context->EndPass()) {
+    VALIDATION_LOG
+        << "Failed to end the current render pass in order to read from "
+           "the backdrop texture and apply an advanced blend.";
+    return nullptr;
+  }
+
+  std::shared_ptr<Texture> input_texture =
+      rendering_config.entity_pass_target->Flip(
+          *renderer.GetContext()->GetResourceAllocator());
+
+  if (!input_texture) {
+    VALIDATION_LOG << "Failed to fetch the color texture in order to "
+                      "apply an advanced blend.";
+    return nullptr;
+  }
+
+  render_passes.push_back(LazyRenderingConfig(
+      renderer, std::move(rendering_config.entity_pass_target)));
+  // Eagerly restore the BDF contents.
+
+  // If the pass context returns a backdrop texture, we need to draw it to the
+  // current pass. We do this because it's faster and takes significantly less
+  // memory than storing/loading large MSAA textures. Also, it's not possible
+  // to blit the non-MSAA resolve texture of the previous pass to MSAA
+  // textures (let alone a transient one).
+  Rect size_rect = Rect::MakeSize(input_texture->GetSize());
+  auto msaa_backdrop_contents = TextureContents::MakeRect(size_rect);
+  msaa_backdrop_contents->SetStencilEnabled(false);
+  msaa_backdrop_contents->SetLabel("MSAA backdrop");
```

Comment:
Should this label have a conditional name depending on caller's use?

Reply:
No because this is the backdrop, its everything that doesn't get filtered - which is conceptually the samer whether you are using an emulated advanced blend or a bdf.